test/hack/resource/pkg/resourcetypes/instanceprofile.go

```diff
@@ -63,10 +58,16 @@ func (ip *InstanceProfile) GetExpired(ctx context.Context, expirationTime time.T
 			continue
 		}
 
-		clusterName, found := lo.Find(out.InstanceProfiles[i].Tags, func(tag iamtypes.Tag) bool {
-			return *tag.Key == k8sClusterTag
+		clusterName, foundClusterName := lo.Find(profiles.Tags, func(tag iamtypes.Tag) bool {
+			return lo.FromPtr(tag.Key) == karpenterTestingTag
 		})
-		if found && slices.Contains(excludedClusters, lo.FromPtr(clusterName.Value)) {
+		// Checking to make sure we are only list resources in the given region
+		region, foundRegion := lo.Find(profiles.Tags, func(tag iamtypes.Tag) bool {
+			return lo.FromPtr(tag.Key) == resourceRegionTag
+		})
+
+		if foundClusterName && slices.Contains(excludedClusters, lo.FromPtr(clusterName.Value)) ||
```

Comment:
So, if we didn't find the region, the region shouldn't be equal to the region (it'll be `""`). The same is true for the excluded clusters. I wonder if we even need the found check here.

Reply:
Yeah, that's a fair point. Removed them